bluemira/equilibria/opt_constraints.py

```diff
@@ -219,3 +302,586 @@ def coil_field_constraints(constraint, vector, grad, eq, B_max, scale):
         grad[:] = dB
     constraint[:] = B - B_max
     return constraint
+
+
+def _get_dummy_equilibrium(equilibrium, I_not_dI):
+    """
+    Get a dummy equilibrium for current optimisation where the background response is
+    solely due to the plasma and passive coils.
+
+    Notes
+    -----
+    When we do dI (current gradient) optimisation, the background vector includes the
```

Comment:
I'm not sure I_not_dI is particularly descriptive where it's used (eg. here where it's controlling whether an equilibrium should be overwritten by a dummy one or not). 

Given how often it crops up - I suspect it may be better to tackle its use with polymorphism of objects setting up arrays for gradient vs absolute current vectors, but that may be beyond the scope of this PR.